search/views.py

```diff
@@ -124,6 +124,11 @@ def do_search(request, course_id=None):
         }
         log.debug(unicode(invalid_err))
 
+    except QueryParseError as parse_error:
```

Comment:
Please add a test case for this.

Reply:
@ciuin Like you did for the course_discovery_search, please add tests to `search/tests/test_view.py` to test these changes.